src/camp_core.F90

```diff
@@ -1592,15 +1735,23 @@ subroutine bin_unpack(this, buffer, pos, comm)
               sub_model_factory%bin_unpack(buffer, pos, l_comm)
     end do
     call camp_mpi_unpack_integer(buffer, pos, this%size_state_per_cell, l_comm)
-    call camp_mpi_unpack_integer(buffer, pos, this%n_cells, l_comm)
     call camp_mpi_unpack_logical(buffer, pos, this%split_gas_aero, l_comm)
     call camp_mpi_unpack_real(buffer, pos, this%rel_tol, l_comm)
     call camp_mpi_unpack_real_array(buffer, pos, this%abs_tol, l_comm)
     call camp_mpi_unpack_integer_array(buffer, pos, this%var_type, l_comm)
-    call camp_mpi_unpack_real_array(buffer, pos, this%init_state, l_comm)
+    call camp_mpi_unpack_real_array(buffer, pos, this%init_state_cell, l_comm)
+
+    allocate(this%init_state(this%size_state_per_cell * this%n_cells))
+    do i_cell = 0, this%n_cells - 1
+      do i_state_elem = 1, this%size_state_per_cell
+        this%init_state(i_state_elem + i_cell * this%size_state_per_cell)=&
+                this%init_state_cell(i_state_elem)
+      end do
+    end do
```

Comment:
Not always. For example, for "camp/data/CAMP_v1_paper/camp_monarch_interface" it is only initialized for MPI rank=0. Then, most of the variables are sent to the rest of the processes. I detect only 1 variable only present in rank=0, which is unique_names (i.e. the names of the species), but it may be more.

Summarizing, "init_state" is initialized at rank=0, then sent to the other ranks